### PR TITLE
Rafactor api controller - add k8s package for k8s objects

### PR DESCRIPTION
**Pull request by renuka-fernando**
Related issue: https://github.com/wso2/k8s-api-operator/issues/326

- Add k8s package
- Remove utils method for creating, getting k8s objects
- Use pointers for memory and perf optimization

### CODE REVIEW

File: api-operator/pkg/k8s/confmap/configmap.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
+package confmap
```
Comment (dinusha92) [inline api-operator/pkg/k8s/confmap/configmap.go:1]: Can you please add the license headers to the new files.

File: api-operator/pkg/k8s/secret/secret.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		logger.Error(err, "error getting secret", "secret", namespaceName)
+		return err
+	}
+
+	// secret already exists and update it
+	return Update(client, secret)
+}
+
+// New returns a new secret object with given namespacedName and data map
+func New(namespacedName types.NamespacedName, data *map[string][]byte, stringData *map[string]string, owner *[]metav1.OwnerReference) *corev1.Secret {
```
Comment (dinusha92) [inline api-operator/pkg/k8s/secret/secret.go:69]: I think we can come up with a better name :). New doesn't give the feeling of a function.

File: api-operator/pkg/k8s/ownerref/owner.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var logger = log.Log.WithName("k8s configmap")
+
+// NewArrayFrom returns an array with a new owner reference object of given meta data
+func NewArrayFrom(typeMeta metav1.TypeMeta, objectMeta metav1.ObjectMeta) *[]metav1.OwnerReference {
```
Comment (dinusha92) [inline api-operator/pkg/k8s/ownerref/owner.go:11]: method name should be a verb which describes the activity.

File: api-operator/pkg/k8s/secret/secret.go (hunk 2 above)
Comment (renuka-fernando) [reply api-operator/pkg/k8s/secret/secret.go:69]: It is like a constructor, construct a k8s secret. But do not create a secret in K8s cluster.
example:
```go
mgwSecret := secret.New(mgwNsName, &mgwData, nil, owner)
```
`secret.Create()` creates a secret in k8s cluster.

**Comment by renuka-fernando** [comment]
Updated the PR with
- Removing k8s sub packages and add generic methods in client.go

Thanks @dinusha92 for you suggestion to remove sub packages

File: api-operator/pkg/controller/api/api_controller.go [hunk 4]
```diff
@@ -236,21 +236,25 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
...
 	operatorOwner, ownerErr := getOperatorOwner(r)
 	if ownerErr != nil {
 		reqLogger.Info("Operator was not found in the " + wso2NameSpaceConst + " namespace. No owner will be set for the artifacts")
 	}
 	userNameSpace := instance.Namespace
 
 	//get configurations file for the controller
-	controlConf, errConf := getConfigmap(r, controllerConfName, wso2NameSpaceConst)
+	controlConf := &corev1.ConfigMap{}
+	errConf := k8s.Get(&r.client, types.NamespacedName{Namespace: wso2NameSpaceConst, Name: controllerConfName}, controlConf)
```
Comment (Shehanir) [inline api-operator/pkg/controller/api/api_controller.go:248]: is this k8s.get method is for creating and deplying configmaps ? Can't we have a more specific name like getConfigmap or deployConfigmap ?

**Pull request merged**